### ISSUE TITLE
Connection refused

### ISSUE THREAD

**Issue by garbo999** [sunspot/sunspot#835]
Any idea why the SOLR server refuses the connection?

I followed the steps in README.md and have the following configuration on my development machine:
- Rails 4.2.5.1
- Ruby 2.1.1p76
- Ubuntu 12.04 LTS

```
saasbook@saasbook:~/Documents/sunspottest$ bundle exec rake sunspot:solr:start
Removing stale PID file at /home/saasbook/Documents/sunspottest/solr/pids/development/sunspot-solr-development.pid
Successfully started Solr ...
```

Now when I run a search from Rails console it looks like the server is not running. I get the following error:

```
2.1.1 :005 > Sentence.search {fulltext 'pizza'}
D, [2016-12-13T12:56:38.512013 #5154] DEBUG -- :   SOLR Request (2.3ms)  [ path=select parameters={fq: ["type:Sentence"], q: "pizza", fl: "* score", qf: "body_text", defType: "edismax", start: 0, rows: 30} ]
RSolr::Error::ConnectionRefused: Connection refused - {:data=>"fq=type%3ASentence&q=pizza&fl=*+score&qf=body_text&defType=edismax&start=0&rows=30", :method=>:post, :params=>{:wt=>:ruby}, :query=>"wt=ruby", :headers=>{"Content-Type"=>"application/x-www-form-urlencoded; charset=UTF-8"}, :path=>"select", :uri=>#<URI::HTTP:0xb5a57d4 URL:http://localhost:8982/solr/development/select?wt=ruby>, :open_timeout=>nil, :read_timeout=>nil, :retry_503=>nil, :retry_after_limit=>nil}
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/rsolr-1.1.2/lib/rsolr/connection.rb:19:in `rescue in execute'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/rsolr-1.1.2/lib/rsolr/connection.rb:14:in `execute'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/rsolr-1.1.2/lib/rsolr/client.rb:181:in `execute'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/rsolr-1.1.2/lib/rsolr/client.rb:175:in `send_and_receive'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/sunspot_rails-2.2.7/lib/sunspot/rails/solr_instrumentation.rb:16:in `block in send_and_receive_with_as_instrumentation'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/notifications.rb:164:in `block in instrument'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/notifications.rb:164:in `instrument'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/sunspot_rails-2.2.7/lib/sunspot/rails/solr_instrumentation.rb:15:in `send_and_receive_with_as_instrumentation'
	from (eval):2:in `post'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/sunspot-2.2.7/lib/sunspot/search/abstract_search.rb:49:in `execute'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/sunspot_rails-2.2.7/lib/sunspot/rails/searchable.rb:389:in `solr_execute_search'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/sunspot_rails-2.2.7/lib/sunspot/rails/searchable.rb:165:in `solr_search'
	from (irb):5
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/railties-4.2.5.1/lib/rails/commands/console.rb:110:in `start'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/railties-4.2.5.1/lib/rails/commands/console.rb:9:in `start'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/railties-4.2.5.1/lib/rails/commands.rb:17:in `<top (required)>'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:274:in `require'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:274:in `block in require'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:274:in `require'
	from /home/saasbook/Documents/sunspottest/bin/rails:9:in `<top (required)>'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:268:in `load'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:268:in `block in load'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /home/saasbook/.rvm/gems/ruby-2.1.1/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:268:in `load'
	from /home/saasbook/.rvm/rubies/ruby-2.1.1/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /home/saasbook/.rvm/rubies/ruby-2.1.1/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from -e:1:in `<main>'
```

**Comment by garbo999** [comment]
Here are a couple more pieces of information:

1. It seems like I should be able to access a SOLR admin panel at `http://localhost:8982/solr/` but it is unavailable.

2. sunspot.yml
```
production:
  solr:
    hostname: localhost
    port: 8983
    log_level: WARNING
    path: /solr/production
    # read_timeout: 2
    # open_timeout: 0.5

development:
  solr:
    hostname: localhost
    port: 8982
    log_level: INFO
    path: /solr/development

test:
  solr:
    hostname: localhost
    port: 8981
    log_level: WARNING
    path: /solr/test
```

Is it possible that I just need to start the daemon?

**Comment by garbo999** [comment]
I figured it out. It was a Java problem.

Note that `bundle exec rake sunspot:solr:run` is more practical because it displays error output. `bundle exec rake sunspot:solr:start` makes it look like the server is running even if there is a Java error in the background.

First, I updated the Java installation on my version of Ubuntu. Not sure if this was necessary (and whether the development kit is required or just the runtime). Here's a good link: https://www.digitalocean.com/community/tutorials/how-to-install-java-on-ubuntu-with-apt-get 

But the main problem was that Java was not in the path. I fixed this as follows (your Java version my vary):

```
export JAVA_HOME=/usr/lib/jvm/java-7-openjdk-i386
```
